diff --git a/arch/arm64/boot/dts/amlogic/g12b_s922x_beelink_gt_king.dts b/arch/arm64/boot/dts/amlogic/g12b_s922x_beelink_gt_king.dts
index 0698ea271222..1170fe006f8c 100644
--- a/arch/arm64/boot/dts/amlogic/g12b_s922x_beelink_gt_king.dts
+++ b/arch/arm64/boot/dts/amlogic/g12b_s922x_beelink_gt_king.dts
@@ -2,6 +2,7 @@
 
 /{
 	model = "Beelink";
+	amlogic-dt-id = "g12b_w400_b";
 	coreelec-dt-id = "g12b_s922x_beelink_gt_king";
 
 	memory@00000000 {
diff --git a/arch/arm64/boot/dts/amlogic/g12b_s922x_beelink_gt_king_rev_a.dts b/arch/arm64/boot/dts/amlogic/g12b_s922x_beelink_gt_king_rev_a.dts
index b8920f530084..beaa36699ad9 100644
--- a/arch/arm64/boot/dts/amlogic/g12b_s922x_beelink_gt_king_rev_a.dts
+++ b/arch/arm64/boot/dts/amlogic/g12b_s922x_beelink_gt_king_rev_a.dts
@@ -2,6 +2,7 @@
 
 /{
 	model = "Beelink";
+	amlogic-dt-id = "g12b_w400_a";
 	coreelec-dt-id = "g12b_s922x_beelink_gt_king_rev_a";
 
 	memory@00000000 {
diff --git a/arch/arm64/boot/dts/amlogic/g12b_s922x_odroid_n2.dts b/arch/arm64/boot/dts/amlogic/g12b_s922x_odroid_n2.dts
index f92f050723ce..f5bc1903832a 100644
--- a/arch/arm64/boot/dts/amlogic/g12b_s922x_odroid_n2.dts
+++ b/arch/arm64/boot/dts/amlogic/g12b_s922x_odroid_n2.dts
@@ -2,7 +2,7 @@
 
 /{
 	model = "Hardkernel ODROID-N2";
-	/delete-property/ amlogic-dt-id;
+	amlogic-dt-id = "g12b_w400_a";
 	coreelec-dt-id = "g12b_s922x_odroid_n2";
 
 	/delete-node/ memory@00000000;
diff --git a/arch/arm64/boot/dts/amlogic/g12b_s922x_odroid_n2plus.dts b/arch/arm64/boot/dts/amlogic/g12b_s922x_odroid_n2plus.dts
index 2e21ae48b6ef..f0e8c4ee527e 100644
--- a/arch/arm64/boot/dts/amlogic/g12b_s922x_odroid_n2plus.dts
+++ b/arch/arm64/boot/dts/amlogic/g12b_s922x_odroid_n2plus.dts
@@ -3,6 +3,7 @@
 
 /{
 	model = "Hardkernel ODROID-N2Plus";
+	amlogic-dt-id = "g12b_w400_c";
 	coreelec-dt-id = "g12b_s922x_odroid_n2plus";
 
 	/delete-node/ memory@00000000;
diff --git a/arch/arm64/boot/dts/amlogic/g12b_s922x_ugoos_am6.dts b/arch/arm64/boot/dts/amlogic/g12b_s922x_ugoos_am6.dts
index bd501c3fd901..a03fa719b390 100644
--- a/arch/arm64/boot/dts/amlogic/g12b_s922x_ugoos_am6.dts
+++ b/arch/arm64/boot/dts/amlogic/g12b_s922x_ugoos_am6.dts
@@ -2,6 +2,7 @@
 
 /{
 	model = "UGOOS AM6";
+	amlogic-dt-id = "g12b_w400_b";
 	coreelec-dt-id = "g12b_s922x_ugoos_am6";
 
 	/delete-node/ memory@00000000;
diff --git a/arch/arm64/boot/dts/amlogic/g12b_s922x_ugoos_am6_rev_a.dts b/arch/arm64/boot/dts/amlogic/g12b_s922x_ugoos_am6_rev_a.dts
index cb74e8f154ed..1496c51873f9 100644
--- a/arch/arm64/boot/dts/amlogic/g12b_s922x_ugoos_am6_rev_a.dts
+++ b/arch/arm64/boot/dts/amlogic/g12b_s922x_ugoos_am6_rev_a.dts
@@ -2,6 +2,7 @@
 
 /{
 	model = "UGOOS AM6";
+	amlogic-dt-id = "g12b_w400_a";
 	coreelec-dt-id = "g12b_s922x_ugoos_am6_rev_a";
 
 	/delete-node/ memory@00000000;
diff --git a/arch/arm64/boot/dts/amlogic/sm1_s905x3_2g_1gbit.dts b/arch/arm64/boot/dts/amlogic/sm1_s905x3_2g_1gbit.dts
index 29a97f56a1f8..73f67ffa921a 100644
--- a/arch/arm64/boot/dts/amlogic/sm1_s905x3_2g_1gbit.dts
+++ b/arch/arm64/boot/dts/amlogic/sm1_s905x3_2g_1gbit.dts
@@ -1,6 +1,7 @@
 #include "sm1_s905x3_2g.dts"
 
 /{
+	amlogic-dt-id = "sm1_ac213_2g";
 	coreelec-dt-id = "sm1_s905x3_2g_1gbit";
 };
 
diff --git a/arch/arm64/boot/dts/amlogic/sm1_s905x3_4g_1gbit.dts b/arch/arm64/boot/dts/amlogic/sm1_s905x3_4g_1gbit.dts
index 4cabbef06909..d090f38196cf 100644
--- a/arch/arm64/boot/dts/amlogic/sm1_s905x3_4g_1gbit.dts
+++ b/arch/arm64/boot/dts/amlogic/sm1_s905x3_4g_1gbit.dts
@@ -1,6 +1,7 @@
 #include "sm1_s905x3_4g.dts"
 
 /{
+	amlogic-dt-id = "sm1_ac213_4g";
 	coreelec-dt-id = "sm1_s905x3_4g_1gbit";
 };
 
